pom.xml

```diff
@@ -0,0 +1,81 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>javacc21</groupId>

```

Comment:
Do you like?

Reply:
It _looks_ okay to me. It's a funny thing. I put some time. (Not a huge amount, but maybe an hour, I dunno) into making it work so that you could open the directory as a project in either Eclipse or IntelliJ and it would just work. However, if there is no need for any of that and it figures it out from Maven, then that is clearly better. 

Have you verified that without the various config files, you can open it as a project in either Eclipse or IJ? Basically, you want to just open the directory as a project in your workspace and, for example in IJ, it's CTRL-F9 to build, but regardless, in either environment, it just builds and one could, in principle, start working immediately. (Though, of course, nobody could just open it and start working immediately because it requires such a huge intellectual investment to understand how everything works, but at least, there is no problem setting up the work environment.)

Reply:
Oh, actually, you can see the file that IJ puts in [here](https://github.com/javacc21/javacc21/blob/master/.idea/ant.xml) and you see that it defines a little hook for the "beforeCompilation" event, which is that it runs the ant parser-gen task. Obviously, I didn't write that file. The GUI generates it. The equivalent one for Eclipse is [here](https://github.com/javacc21/javacc21/blob/master/.externalToolBuilders/Parser-gen.launch) Again, that was generated by Eclipse, but if you do a clean build (or an incremental one too, I think) it runs the parser-gen task. I only figured out how to do this in Eclipse recently. I found the way to do it in IntelliJ much more quickly.

But again, if both Eclipse and IJ can deduce these things from the Maven settings, then that is much better. I'm not utterly confident about checking in these Eclipse/IJ files, since there is no guarantee about the format being stable over time and so on. Well, besides, in general, this is not the kind of thing I want to spend very much time with, so if it can just be specied in Maven and just work, that is better obviously.